scalalib/src/mill/scalalib/SemanticDbJavaModule.scala

```diff
@@ -72,45 +99,98 @@ trait SemanticDbJavaModule extends JavaModule { hostModule =>
       }
   }
 
-  def semanticDbData: T[PathRef] = hostModule match {
-    // TODO: also generate semanticDB for Java and Mixed-Java projects
-    case m: ScalaModule =>
-      T.persistent {
-        val sv = m.scalaVersion()
-
-        val scalacOptions = (
-          m.allScalacOptions() ++
-            semanticDbEnablePluginScalacOptions() ++ {
-              if (ZincWorkerUtil.isScala3(sv)) {
-                Seq("-Xsemanticdb")
-              } else {
-                Seq(
-                  "-Yrangepos",
-                  s"-P:semanticdb:sourceroot:${T.workspace}",
-                  "-Ystop-after:semanticdb-typer"
-                )
+  private def resolvedJavaSemanticDbPluginIvyDeps: Target[Agg[PathRef]] = T {
+    resolveDeps(T.task { javaSemanticDbPluginIvyDeps().map(bindDependency()) })()
+  }
+
+  def semanticDbData: T[PathRef] = {
+    // TODO: add extra options for Java 17+
+    def javacOptionsTask(m: JavaModule): Task[Seq[String]] = T.task {
+      m.javacOptions() ++ Seq(
+        s"-Xplugin:semanticdb -sourceroot:${T.workspace} -targetroot:${T.dest / "classes"}"
```

Comment:
So if I'm not mistaken the issue you're seeing with:

```
java.nio.file.FileSystemNotFoundException: Provider "vf" not installed
```

Stems from not including `-build-tool:mill` here. I hit this as well a while back and it caused me to create https://github.com/sourcegraph/scip-java/pull/497/files#diff-0e7ae96d79f653d2c89b1b6e6798bffa6b4effd046f340e96ed562a3b15bde4c. But that option is used to correctly set https://github.com/sourcegraph/scip-java/blob/3725605c81a08d04a27c7939e818a091df35e93e/semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbTaskListener.java#L103-L129 if I remember correctly.

Reply:
😅 I just realized I'm still using `-build-tool:sbt` in mine:

https://github.com/ckipp01/mill-scip/blob/main/plugin/src/io/kipp/mill/scip/Scip.scala#L200

Because that was before I made the changes in `scip-java`.

Reply:
Thank you. Looks like `-build-tool:mill` is not working, but `-build-tool:sbt` is.

```
[error] /home/lefou/work/opensource/mill/scratch-hello-mixed/java/src/example/HelloJava.java:1:1: semanticdb-javac: unknown flag '-build-tool:mill'
```

I use version `0.8.9`.

Also, I noticed, that the semanticDB for Java is placing the files under `META-INF/semanticdb/<full-root-path>. A bit strange and huge potential to fail on Windows.

Reply:
>. I use version 0.8.9.

ugh... you just made me realize I used `args.equal` instead of `arg.equal`. I'll need to fix that upstream.

> Also, I noticed, that the semanticDB for Java is placing the files under META-INF/semanticdb/_full-root-path_. A bit strange and huge potential to fail on Windows.

Ahhh, it seems that windows testing is commented out in scip-java. I'd maybe recommend creating an issue there.

Reply:
>. I use version 0.8.9.

ugh... you just made me realize I used `args.equal` instead of `arg.equal`. I'll need to fix that upstream.

> Also, I noticed, that the semanticDB for Java is placing the files under META-INF/semanticdb/_full-root-path_. A bit strange and huge potential to fail on Windows.

Ahhh, it seems that windows testing is commented out in scip-java. I'd maybe recommend creating an issue there.

Reply:
I still think I'm using it wrong or it is a bug in semanticdb-java, as the full path should actually only be the package hierarchy.